api/admin/controller/__init__.py

```diff
@@ -2223,7 +2223,7 @@ def _set_integration_setting(self, integration, setting):
                     "The configuration value for %(setting)s is invalid.",
                     setting=setting.get("label"),
                 ))
-        if not value and setting.get("required"):
+        if not value and setting.get("required") and not setting.get("default"):
             return INCOMPLETE_CONFIGURATION.detailed(
                 _("The configuration is missing a required setting: %(setting)s",
                   setting=setting.get("label")))
```

Comment:
"Are all the required settings set?" is a good candidate for moving into validate_formats next. I noticed you added another check that returns INCOMPLETE_CONFIGURATION and this would let you remove all of those.